vyper/parser/self_call.py

```diff
@@ -44,10 +45,10 @@ def make_call(stmt_expr, context):
             getpos(stmt_expr)
         )
 
-    if sig.private:
-        return call_self_private(stmt_expr, context, sig)
-    else:
-        return call_self_public(stmt_expr, context, sig)
```

Comment:
I have a maintenance concern here which is that this will leave a bunch of dead code in the codebase. @jacqueswww @davesque thoughts?

Reply:
There is significant overlap between the removed function and `external_contract_call` in `vyper/parser/external_call.py`

Reply:
Well looking at `call_self_public` = it doesn't really have that much dead code or functions it calls out to? Most of those functions are used by `call_self_private` from what I checked.